pdfarranger/pdfarranger.py

```diff
@@ -1669,18 +1665,31 @@ def sw_scroll_event(self, _scrolledwindow, event):
     def zoom_set(self, level):
         """Sets the zoom level"""
         level = min(max(level, -10), 40)
-        if level == self.zoom_level:
+        zoom_level_old = self.zoom_level
+        self.zoom_level = level
+        self.zoom_scale = 0.2 * (1.1 ** level)
+        # Limit max zoom level so that thumbnail is max 23Mb
+        if len(self.model) > 0:
+            max_limit = 6000000  # 6000000 pixels * 4 byte/pixel -> 23Mb
+            max_page_size = max(p.size[0] * p.size[1] * p.scale ** 2 for p, _ in self.model)
```

Comment:
`p.size` is the original size without crops. Do we need `p.size_in_points()` here?

Reply:
The thumbnails are always rendered in full size and crops are applied later. Goal here is to limit memory usage, so I would say this is how it need to be for now. May be that we should change how the rendering work so it would only render the cropped page. (I did a test now where I opened a pdf with many pages, added crop 99% to left and top, and zoomed all out. Memory usage was then 12Gb..) But I'd say that is for a new issue. The 6000000 limit could be higher. Do you think it should be?

Reply:
I think it's perfectly fine to optimize for the general/expected use case. If someone really needs 99% crop and this becomes a problem I'd ask them nicely to use a different tool...

I haven't gone into the details of what you are doing here but from what I understand it sounds like great work, thanks! 👍

Reply:
Thanks for the explanation. I see the primary use case of first cropping and then rendering in the support of negative crops, but that is probably quite a bit into the future.